modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/SemanticRequestMetrics.java

```diff
@@ -24,35 +24,60 @@
 import com.codahale.metrics.Histogram;
 import com.codahale.metrics.Meter;
 import com.codahale.metrics.RatioGauge;
+import com.codahale.metrics.RatioGauge.Ratio;
+import com.codahale.metrics.Timer;
+import com.spotify.apollo.Request;
+import com.spotify.apollo.Response;
 import com.spotify.apollo.StatusType;
 import com.spotify.apollo.metrics.RequestMetrics;
-import com.spotify.apollo.metrics.TimerContext;
 import com.spotify.metrics.core.MetricId;
 import com.spotify.metrics.core.SemanticMetricRegistry;
 
+import java.util.Optional;
+import java.util.Set;
 import java.util.function.Supplier;
 
+import okio.ByteString;
+
 import static com.spotify.apollo.StatusType.Family.INFORMATIONAL;
 import static com.spotify.apollo.StatusType.Family.SUCCESSFUL;
+import static com.spotify.apollo.metrics.semantic.Metric.DROPPED_REQUEST_RATE;
+import static com.spotify.apollo.metrics.semantic.Metric.ENDPOINT_REQUEST_DURATION;
+import static com.spotify.apollo.metrics.semantic.Metric.ENDPOINT_REQUEST_RATE;
+import static com.spotify.apollo.metrics.semantic.Metric.ERROR_RATIO;
+import static com.spotify.apollo.metrics.semantic.Metric.REPLY_SIZE;
+import static com.spotify.apollo.metrics.semantic.Metric.REQUEST_FANOUT_FACTOR;
+import static com.spotify.apollo.metrics.semantic.Metric.REQUEST_SIZE;
 import static java.util.Objects.requireNonNull;
 
+// Optional fields are fine; they enable the use of the 'ifPresent' idiom which is more readable
+// than if statements
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
```

Comment:
Good idea; I think once we've got our checkstyle config open-sourced, we can start addressing that. I added it to avoid IntelliJ warnings, I wonder if it picks up checkstyle suppressions config, do you know?